os/fs/driver/mtd/smart.c

```diff
@@ -4059,30 +4062,67 @@ static int smart_validate_crc(FAR struct smart_struct_s *dev)
 	header = (FAR struct smart_sect_header_s *)dev->rwbuffer;
 
 #ifdef CONFIG_SMART_CRC_16
-
 	/* Test 16-bit CRC. */
 
 	if (crc != *((uint16_t *)header->crc16) && crc != SMART_ERASEDSTATE_CRC) {
 		return -EIO;
 	}
 #elif defined(CONFIG_SMART_CRC_32)
-
 	if (crc != *((uint32_t *)header->crc32)) {
 		return -EIO;
 	}
-#else
-	/* Test 8-bit CRC for CRC8 & basic case for smart_scan. */
+#elif defined(CONFIG_SMART_CRC_8)
+	/* Test 8-bit CRC for CRC8. */
+
 	if (crc != header->crc8 && crc != SMART_ERASEDSTATE_CRC) {
 		return -EIO;
 	}
+#else
+	/*CONFIG_MTD_SMART_ENABLE_CRC not set, perform minimal CRC check
```

Comment:
As I told you before,  below can't be acceptable
```
crc != SMART_ERASEDSTATE_CRC 
```
Bit flip can be occured during the power down case.

Reply:
- Yes sir, agreed, with reference to your inputs on the previous PR, the functioning of the CRC check (smart_validate_crc function) and the **hard coding for the pre-build of smartfs.bin has been separated from on-device functioning** by use of NXFUSE_HOST_BUILD flag.
- The hard coded values are **used only when smartfs.bin image is being created on the host(Linux) machine** where abrupt power outages may not pose a problem.
- SMART_ERASEDSTATE_CRC **does not come into play at all** when smartfs operations are being carried out **on the device.**